src/component/AddItem.js

```diff
@@ -3,27 +3,47 @@ import { withFirestore } from "react-firestore";
 import "../App.css";
 import Navbar from "./Navbar";
 import BackButton from "./BackButton";
+import DuplicateMessage from "./DuplicateMessage";
 
 const AddItem = ({ firestore }) => {
   const [name, setName] = useState("");
-  const [frequency, setFrequency] = useState("14");
+  const [numberOfDays, setNumberOfDays] = useState("14");
+  const [duplicate, setDuplicate] = useState(false);
+
   const uniqueToken = localStorage.getItem("uniqueToken");
 
   //   Write item to Firebase setting uniqueToken as document name
-  const addItem = (name, frequency) => {
+  const addItem = (normalizedName, numberOfDays) => {
+    // adds new items collection to database
     firestore
       .collection("lists")
       .doc(uniqueToken)
       .set({ items: "" });
 
-    firestore
+    // reference path to specific document from items collection
+    // sets document ID equal to item name
+    const itemsDocRef = firestore
       .collection("lists")
       .doc(uniqueToken)
       .collection("items")
-      .add({
-        name: name,
-        frequency: +frequency
-      });
+      .doc(normalizedName);
+
+    // checks whether an existing doc ID is equal to new item name
+    itemsDocRef.get().then(docSnapshot => {
+      if (docSnapshot.exists) {
+        itemsDocRef.onSnapshot(doc => {
+          setDuplicate(true);
+          setTimeout(function() {
+            setDuplicate(false);
+          }, 2500);
```

Comment:
Tip for the future: As a reviewer, I don't understand why you chose `2500` at first hand, a way to fix this is:
- Add a meaningful comment about that 2500
- Create a variable with a meaningful name (it doesn't matter if it's a long name, our IDEs could handle that!) and assign 2500 to it.

It could be something like

```js
const TIME_WINDOW_BEFORE_REFRESH = 2500;

setTimeOut(...., TIME_WINDOW_BEFORE_REFRESH);
```